src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
```

Comment:
The 'ensureCapacity' method name is a bit misleading because it implies the capacity is always ensured. However, inside the method, there is a condition that may not lead to an increase in capacity. Consider renaming the method to reflect that capacity is only ensured if necessary, for example 'growIfNecessary' or 'ensureCapacityIfFull'.